plugins/webp-uploads/settings.php

```diff
@@ -210,19 +199,70 @@ function webp_uploads_use_picture_element_callback(): void {
+				?>
+				<input
+					type="hidden"
+					id="webp_uploads_use_picture_element_hidden"
+					name="webp_uploads_use_picture_element"
+					value="1"
+				>
+				<?php
+			}
+			esc_html_e( 'Use <picture> Element', 'webp-uploads' );
+			?>
 			<em><?php esc_html_e( '(experimental)', 'webp-uploads' ); ?></em>
 		</label>
 		<p class="description" id="webp_uploads_use_picture_element_description"><?php esc_html_e( 'The picture element serves a modern image format with a fallback to JPEG. Warning: Make sure you test your theme and plugins for compatibility. In particular, CSS selectors will not match images when using the child combinator (e.g. figure > img).', 'webp-uploads' ); ?></p>
-		<div id="webp_uploads_jpeg_fallback_notice" class="notice notice-info inline" <?php echo $picture_element_enabled ? '' : 'hidden'; ?>>
+		<div id="webp_uploads_jpeg_fallback_notice" class="notice notice-info inline" <?php echo $picture_element_option ? '' : 'hidden'; ?>>
 			<p><?php esc_html_e( 'Picture elements will only be used when JPEG fallback images are available. So this will not apply to any images you may have uploaded while the "Also generate JPEG" setting was disabled.', 'webp-uploads' ); ?></p>
 		</div>
 	</div>
 	<script>
 		document.getElementById( 'webp_uploads_use_picture_element' ).addEventListener( 'change', function () {
 			document.getElementById( 'webp_uploads_jpeg_fallback_notice' ).hidden = ! this.checked;
 		} );
+
+		// Listen for clicks on the JPEG output checkbox, enabling/disabling the
+		// picture element checkbox accordingly.
+		document.getElementById( 'perflab_generate_webp_and_jpeg' ).addEventListener( 'change', function () {
+			document.querySelector( '.webp-uploads-use-picture-element' ).classList.toggle( 'webp-uploads-disabled', ! this.checked );
+			document.getElementById( 'webp_uploads_picture_element_notice' ).hidden = this.checked;
+			document.getElementById( 'webp_uploads_picture_element_fieldset' ).classList.toggle( 'disabled', ! this.checked );
+
+			var checkbox = document.getElementById( 'webp_uploads_use_picture_element' );
+			checkbox.disabled = ! this.checked;
+
+			// Remove or inject hidden input to preserve original setting value as needed.
+			if ( this.checked ) {
+				var hiddenInput = document.getElementById( 'webp_uploads_use_picture_element_hidden' );
+				if ( hiddenInput ) {
+					hiddenInput.parentElement.removeChild( hiddenInput );
+				}
+			} else if ( checkbox.checked && ! document.getElementById( 'webp_uploads_use_picture_element_hidden' ) ) {
+				// The hidden input is only needed if the value was originally set (i.e. the checkbox enabled).
+				var hiddenInput = document.createElement( 'input' );
+				hiddenInput.setAttribute( 'type', 'hidden' );
+				hiddenInput.setAttribute( 'id', 'webp_uploads_use_picture_element_hidden' );
+				hiddenInput.setAttribute( 'name', checkbox.getAttribute( 'name' ) );
+				hiddenInput.setAttribute( 'value', checkbox.getAttribute( 'value' ) );
```

Comment:
Less verbose if properties are used instead:

```suggestion
				hiddenInput.type = 'hidden';
				hiddenInput.id = 'webp_uploads_use_picture_element_hidden';
				hiddenInput.name = checkbox.name;
				hiddenInput.value = checkbox.value;
```

Reply:
Does this do the same? I believe I had run into weird issues in the past, where one is not like the other, i.e. not in sync. But maybe that was a specific situation. Are you aware of any quirks with attributes vs properties in that sense?

Reply:
Yeah, although in this case they will be equivalent. We're using the properties elsewhere in this file (e.g. `hidden`), so I think also good for consistency.

Reply:
Done in ae2f0c721e8a422e2bf881f0b1d7ce96680e3705